src/services/member/plugins/storage/service.ts

```diff
@@ -0,0 +1,39 @@
+import { FileItemType } from '@graasp/sdk';
+
+import { Repositories } from '../../../../utils/repositories';
+import { DEFAULT_MAX_STORAGE } from '../../../item/plugins/file/utils/constants';
+import { StorageExceeded } from '../../../item/plugins/file/utils/errors';
+import { Member } from '../../entities/member';
+
+export class StorageService {
+  getMaximumStorageSize(_actor: Member): number {
+    // todo: depends on user/group
+    return DEFAULT_MAX_STORAGE;
+  }
+
+  async getStorageLimits(actor: Member, type: FileItemType, { itemRepository }: Repositories) {
+    return {
+      current: await itemRepository.getItemSumSize(actor?.id, type),
+      maximum: this.getMaximumStorageSize(actor),
+    };
+  }
+
+  // check the user has enough storage to create a new item given its size
+  // get the complete storage
+  // todo: include more item types
+  async checkRemainingStorage(
+    actor: Member,
+    type: FileItemType,
```

Comment:
Is it necessary to pass the item type to the storage service? Shouldn't it be agnostic and only know about sizes?